sources/rules.c

```diff
@@ -346,6 +347,63 @@ od_rule_t *od_rules_forward(od_rules_t *rules, char *db_name, char *user_name,
 	return rule_default_default_default;
 }
 
+static od_rule_t *
+od_rules_forward_sequential(od_rules_t *rules, char *db_name, char *user_name,
```

Comment:
maybe we can rewrite od_rules_forward logic to handle `sequential` inside of it? Im not a big fan of rewriting core logic functions twice

Reply:
> maybe we can rewrite `od_rules_forward` logic to handle `sequential` inside of it (consider macros if different signature needed)? Im not a big fan of rewriting core logic functions twice

But `od_rules_forward_sequential` significantly differs from old `od_rules_forward`, I think two separate functions for old and new rule matching algorithms are more clear